tests/components/omnilogic/test_config_flow.py

```diff
@@ -0,0 +1,139 @@
+            result["flow_id"],
+            DATA,
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "Omnilogic"
+    assert result2["data"] == DATA
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_already_configured(hass):
+    """Test config flow when Omnilogic component is already setup."""
+    MockConfigEntry(domain="omnilogic", data=DATA).add_to_hass(hass)
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "abort"
+
+
+async def test_with_invalid_credentials(hass):
+    """Test with invalid credentials."""
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.omnilogic.OmniLogic.connect",
+        side_effect=LoginException,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            DATA,
+        )
+
+
+async def test_form_cannot_connect(hass):
+    """Test if invalid response or no connection returned from Hayward."""
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.omnilogic.OmniLogic.connect",
+        side_effect=OmniLogicException,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            DATA,
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_with_unknown_error(hass):
+    """Test with unknown error response from Hayward."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.omnilogic.OmniLogic.connect",
+        side_effect=Exception,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            DATA,
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "unknown"}
+
+
+async def test_option_flow(hass):
+    """Test option flow."""
+    entry = MockConfigEntry(domain=DOMAIN, data=DATA)
+    entry.add_to_hass(hass)
+
+    assert not entry.options
+
+    result = await hass.config_entries.options.async_init(
```

Comment:
We need to set up the config entry before starting the options flow. Use `hass.config_entries.async_setup`. Make sure to patch the integration setup functions to avoid setting up the complete integration.

Reply:
OK - have added full setup with patch instead of the Mock setup.